platform/ios/src/MGLMapViewDelegate.h

```diff
@@ -29,9 +30,25 @@ NS_ASSUME_NONNULL_BEGIN
  
  @param mapView The map view whose viewpoint will change.
  @param animated Whether the change will cause an animated effect on the map.
+
+ @note If `mapView:regionWillChangeAnimated:reason` is implemented this method will not be called.
  */
 - (void)mapView:(MGLMapView *)mapView regionWillChangeAnimated:(BOOL)animated;
 
+/**
+ Tells the delegate that the viewpoint depicted by the map view is about to change.
+
+ This method is called whenever the currently displayed map camera will start
+ changing for any reason.
+
+ @param mapView The map view whose viewpoint will change.
+ @param animated Whether the change will cause an animated effect on the map.
+ @param reason The reason for the camera change.
+
+ @note If this method is implemented `mapView:regionWillChangeAnimated:` will not be called.
+ */
+- (void)mapView:(MGLMapView *)mapView regionWillChangeAnimated:(BOOL)animated reason:(MGLCameraChangeReason)reason;
```

Comment:
`-mapView:regionWillChangeAnimated:` is already a bit awkward grammatically, but only because there was nothing to stick between “change” and “animated”. This method name would read better as `-mapView:regionWillChangeForReason:animated:`.

Reply:
Love this - I was struggling with the naming, and this makes far more sense.

Reply:
Wouldn’t regionWillChangeWithReason: be more idiomatic? I believe the cocoa API style prefers “with” over “for”, but I could go either way.

Reply:
One reason to switch to "for" is for the improved auto-translation to the swift method name.
i.e. going from 
`func mapView(_ mapView: MGLMapView, regionIsChangingWithReason reason: MGLCameraChangeReason)`
to 
`func mapView(_ mapView: MGLMapView, regionIsChangingFor reason: MGLCameraChangeReason)`

But I'm happy with either. What's more common?

Reply:
I think the token "with" is more appropriate here, since we're providing the receiver with a value representing the reason. 

The ergonomics of the swift bridging depends on the readability of the `MGLMapChangeReason` type names (and taste, I suppose) but I still have a mild preference for "with" in either language.

Reply:
Some Apple APIs use `withReason:` (NSProcessInfo) while others use `forReason:` (Intents). My impression is that the APIs introduced post-Swift 1.0 have tended to go with `forReason:`. I think `for: reason` reads better at the call site than `with: reason`, but I don’t feel strongly about that.